pkg/controller/controller_binding.go

```diff
@@ -414,9 +460,18 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 	// All updates not having a DeletingTimestamp will have been handled above
 	// and returned early. If we reach this point, we're dealing with an update
 	// that's actually a soft delete-- i.e. we have some finalization to do.
+	return c.reconcileServiceInstanceCredentialDelete(binding, toUpdate, instance)
+}
+
+func (c *controller) reconcileServiceInstanceCredentialDelete(binding *v1alpha1.ServiceInstanceCredential, toUpdate *v1alpha1.ServiceInstanceCredential, instance *v1alpha1.ServiceInstance) error {
```

Comment:
I would rather not split the reconciliation method like this.  I think the reconciliation loop should set the status indicating that orphan mitigation is required and then begin orphan mitigation in the next reconcilation.  There is not a need to begin orphan mitigation right away once you detect that it needs to be done and it will make the logic simpler.

What will need to change about this PR (aside from undoing the method extraction) is that the test to see whether the unbind flow needs to be run will have to account for orphan mitigation now as a cause in addition to the deletion timestamp being set.

Reply:
>  I think the reconciliation loop should set the status indicating that orphan mitigation is required and then begin orphan mitigation in the next reconcilation.

That's ultimately the flow of what @jpeeler wrote. It calls `c.setOrphanMitigation`, returns, and then the next reconciliation is when the mitigation begins.

Reply:
> I would rather not split the reconciliation method like this.

I think splitting it `c.reconcileServiceInstanceCredential` into a separate delete is absolutely a good thing and is exactly what the existing logic in instances does. The single reconciliation function for SICs is right now very unwieldy.

Whether we have a separate `doOrphanMitigation` function is another question, but at the very least I'd like to the `reconcileServiceInstanceCredentialDelete` stay.

Reply:
If we'd like to hold off on any refactoring until we're feature complete, though, I can understand that.